### PR TITLE
Bump bs-platform from 5.2.0 to 7.0.1

### PR DESCRIPTION
**Pull request by imbsky**

### CODE REVIEW

**Comment by MoOx** [comment]
Hi @imbsky, thanks for raising this.
What would be nice while we are doing this upgrade is to also simplify all Js.t types like this https://github.com/reason-react-native/reason-react-navigation/blob/c76161627d9986d86e6abcadfcfa7e15bb1d4cd8/src/Navigation.re#L4-L11
to remove them since now records are encoded as js objects!

Something similar has been started for reason-react-native: https://github.com/reason-react-native/reason-react-native/pull/631

**Comment by imbsky** [comment]
Hi @MoOx, I made some changes. I'm not used to BuckleScript, is there anything else I should change?

File: src/AppContainer.re [hunk 1]
```diff
@@ -34,7 +33,7 @@ external makeProps:
     ~persistNavigationState: persistNavigationState=?,
     ~loadNavigationState: loadNavigationState=?,
     ~screenProps: 'screenProps=?,
-    ~ref: Js.Nullable.t(NavigationContainer.t) => unit=?,
+    ~ref: option(NavigationContainer.t) => unit=?,
```
Comment (MoOx) [inline src/AppContainer.re]: Not sure that's safe. nullable is for null, option None is encoded as undefined, which is != null.

Comment (imbsky) [reply src/AppContainer.re]: Okay, I'll fix it.

**Comment by imbsky** [comment]
Sorry for the late response, I just updated it!

File: src/Navigation.re [hunk 2]
```diff
@@ -2,12 +2,12 @@ open ReactNative;
...
-  .
-  "action": NavigationActions.action,
-  "context": string,
-  "type": string,
-  "state": NavigationState.t,
-  "lastState": Js.Nullable.t(NavigationState.t),
+  action: NavigationActions.action,
+  context: string,
+  [@bs.as "type"]
+  _type: string,
```
Comment (imbsky) [inline src/Navigation.re]: @MoOx I wonder if there is no problem with this.

**Comment by MoOx** [comment]
I have made a similar change to v5 branch but I named it v5-bs7.
Not sure how we want to got with the 5/7 strategy (following https://github.com/reason-react-native/reason-react-native/pull/631)...

**Pull request closed without merging**